hi-ml-azure/src/health_azure/utils.py

```diff
@@ -1803,31 +1807,79 @@ def aggregate_hyperdrive_metrics(
+        run = get_aml_run_from_run_id(run_id, aml_workspace=workspace)
     assert isinstance(run, HyperDriveRun)
     metrics: DefaultDict = defaultdict()
-    for child_run in run.get_children():
+    for child_run in run.get_children():  # type: ignore
         child_run_metrics = child_run.get_metrics()
+        keep_metrics = keep_metrics or child_run_metrics.keys()
+
         child_run_tag = get_tags_from_hyperdrive_run(child_run, child_run_arg_name)
-        for k, v in child_run_metrics.items():
-            if k not in metrics:
-                metrics[k] = {}
-            metrics[k][child_run_tag] = v
+        for metric_name, metric_val in child_run_metrics.items():
+            if metric_name in keep_metrics:
+                if metric_name not in metrics:
+                    metrics[metric_name] = {}
+                metrics[metric_name][child_run_tag] = metric_val
 
     df = pd.DataFrame.from_dict(metrics, orient="index")
     return df
 
 
+def get_metrics_for_childless_run(
+    run_id: Optional[str] = None,
+    run: Optional[Run] = None,
+    keep_metrics: Optional[List[str]] = None,
+    aml_workspace: Optional[Workspace] = None,
+    workspace_config_path: Optional[Path] = None,
+) -> pd.DataFrame:
+    """
+    For a given Run object or id, retrieves the metrics from that Run and returns them as a pandas DataFrame.
+    Optionally filters the metrics logged in the Run, by providing a list of metrics to keep. This function
+    expects a childless AML Run. If you wish to aggregate metrics for a Run with children (i.e. a HyperDriveRun),
+    please use the function 'aggregate_hyperdrive_metrics'.
+
+    :param run: A (childless) Run object to retrieve the metrics from. Either this or run_id must be provided
+    :param run_id: The id (type: str) of a (childless) AML Run. Either this or run must be provided.
+    :param keep_metrics: An optional list of metric names to filter the returned metrics by
+    :param aml_workspace: If run_id is provided, this is an optional AML Workspace object to retrieve the Run from
+    :param workspace_config_path: If run_id is provided, this is an optional path to a config containing details of the
+        AML Workspace object to retrieve the Run from.
+    :return: A Pandas DataFrame containing the metrics
+    """
+    if run is None:
+        assert run_id is not None, "Either run or run_id must be provided"
+        workspace = get_workspace(aml_workspace=aml_workspace, workspace_config_path=workspace_config_path)
+        run = get_aml_run_from_run_id(run_id, aml_workspace=workspace)
+    if isinstance(run, _OfflineRun):
+        logging.warning("Can't get metrics for _OfflineRun object")
+        return pd.DataFrame({})
+    metrics = {}
+    run_metrics = run.get_metrics()  # type: ignore
+    keep_metrics = keep_metrics or run_metrics.keys()
+    for metric_name, metric_val in run_metrics.items():
+        if metric_name in keep_metrics:
+            metrics[metric_name] = metric_val
```

Comment:
There is some amount of code duplication here with the hyperdrive equivalent.

Reply:
There is a bit, but it's not trivial to replace since the hyperdrive equivalent builds a default dict with nested levels (metric name and then child index - e.g. `{'test/auoc': {'0': 0.9, '1': 0.1}}`) so I propose that this can be dealt with in a future PR if necessary?